packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -83,8 +83,19 @@ export default class Proxy {
     return id
   }
 
-  unregisterProxy(id) {
-    return this._db.remove(id)
+  async unregisterProxy(id) {
+    const { vmUuid } = await this._getProxy(id)
+
+    await this._db.remove(id)
+
+    if (vmUuid !== undefined) {
+      await this._app
+        .unbindLicense({
+          boundObjectId: vmUuid,
+          productId: this._xoProxyConf.licenseProductId,
+        })
+        .catch(log.warn)
```

Comment:
IMHO, `await` + `catch()` is not obvious, we should add a comment explaining why we wait.